aries/aries_vcx_core/src/wallet/askar/mod.rs

```diff
@@ -0,0 +1,194 @@
+mod askar_record_wallet;
+mod askar_utils;
+mod crypto_box;
+mod entry;
+mod entry_tags;
+mod packing;
+mod packing_types;
+mod rng_method;
+mod sig_type;
+
+#[derive(Debug)]
+pub struct AskarWallet {
+    backend: Store,
+    profile: Option<String>,
+}
+
+impl BaseWallet for AskarWallet {}
+
+impl AskarWallet {
+    pub async fn create(
+        db_url: &str,
+        key_method: StoreKeyMethod,
+        pass_key: PassKey<'_>,
+        recreate: bool,
+        profile: Option<String>,
+    ) -> Result<Self, AriesVcxCoreError> {
+        let backend =
+            Store::provision(db_url, key_method, pass_key, profile.clone(), recreate).await?;
+
+        Ok(Self { backend, profile })
+    }
+
+    pub async fn open(
+        db_url: &str,
+        key_method: Option<StoreKeyMethod>,
+        pass_key: PassKey<'_>,
+        profile: Option<String>,
+    ) -> Result<Self, AriesVcxCoreError> {
+        Ok(Self {
+            backend: Store::open(db_url, key_method, pass_key, profile.clone()).await?,
+            profile,
+        })
+    }
+
+    async fn fetch_local_key(
+        &self,
+        session: &mut Session,
+        key_name: &str,
+    ) -> VcxCoreResult<LocalKey> {
+        Ok(self
+            .fetch_key_entry(session, key_name)
+            .await?
+            .load_local_key()?)
+    }
+
+    async fn fetch_key_entry(
+        &self,
+        session: &mut Session,
+        key_name: &str,
+    ) -> Result<KeyEntry, AriesVcxCoreError> {
+        session.fetch_key(key_name, false).await?.ok_or_else(|| {
+            AriesVcxCoreError::from_msg(
+                AriesVcxCoreErrorKind::WalletRecordNotFound,
+                format!("no key with name '{}' found in wallet", key_name),
+            )
+        })
+    }
+
+    async fn insert_key(
+        &self,
+        session: &mut Session,
+        alg: KeyAlg,
+        seed: &[u8],
+        rng_method: RngMethod,
+    ) -> Result<(String, LocalKey), AriesVcxCoreError> {
+        let key = LocalKey::from_seed(alg, seed, rng_method.into())?;
+        let key_name = local_key_to_bs58_name(&key)?;
+        session
+            .insert_key(&key_name, &key, None, None, None)
+            .await?;
+        Ok((key_name, key))
+    }
+
+    async fn find_did(
+        &self,
+        session: &mut Session,
+        did: &str,
+        category: &str,
+    ) -> VcxCoreResult<Option<DidData>> {
+        if let Some(entry) = session.fetch(category, did, false).await? {
+            if let Some(val) = entry.value.as_opt_str() {
+                return Ok(Some(serde_json::from_str(val)?));
+            }
+        }
+
+        Ok(None)
+    }
+
+    async fn find_current_did(
+        &self,
+        session: &mut Session,
+        did: &str,
+    ) -> VcxCoreResult<Option<DidData>> {
+        self.find_did(session, did, DID_CATEGORY).await
+    }
+
+    async fn insert_did(
+        &self,
+        session: &mut Session,
+        did: &str,
+        category: &str,
+        verkey: &str,
```

Comment:
The callsites (albeit these are just internal) are required to know, that verkey is base58 encoded. Perhaps we could name this more explicitly `create_did_from_local_key` and accept `local_key: LocalKey`  instead of `verkey: &str`.

Reply:
Changed to `public_key::Key` as a part of fix for key name above.